src/components/Dialog-Window/Notes/DialogNote.tsx

```diff
@@ -10,14 +10,38 @@ export type NoteProps = {
 export const Note: React.FC<NoteProps> = ({ maxLength, id }) => {
   const [userData, setUserData] = useLocalStorage(id);
   const [note, setNote] = React.useState(userData[id].note ?? "");
+  const [dynamicSavingText, setDynamicSavingText] = React.useState("");
+  const [savingTextTimeout, setSavingTextTimeout] =
+    React.useState<NodeJS.Timeout>();
+
+  // TODO: Translate
+  const savingTextLabel = "lagring...";
+  const savedTextLabel = "lagret";
+
+  const setSavingText = (): void => {
+    setDynamicSavingText(savingTextLabel);
+    setSavingTextTimeout(
+      setTimeout(() => {
+        const timestamp = new Date();
+        const hours = String(timestamp.getHours()).padStart(2, "0");
+        const minutes = String(timestamp.getMinutes()).padStart(2, "0");
+        setDynamicSavingText(`${savedTextLabel} ${hours}:${minutes}`);
+      }, 650),
+    );
+  };
 
   React.useEffect(() => {
     // TODO: If this becomes laggy, add a debounce-timer to avoid saving more often than, say, every 100ms.
     userData[id].note = note;
     setUserData(userData);
-  }, [userData, id, note, setUserData]);
+    // ensure there's no memory leak on component unmount during timeout
+    return () => {
+      if (savingTextTimeout) clearTimeout(savingTextTimeout);
```

Comment:
There's is a minor chance that `savingTextTimeout` might be `0`, which is falsy. It's better to check for nullish values.

```suggestion
      if (savingTextTimeout != null) clearTimeout(savingTextTimeout);
```

Reply:
It's really good that we're clearing timeouts before unmount, though! Great catch 🎉

Reply:
✅